charts/keycloak/README.md

```diff
@@ -24,217 +24,289 @@ By default, the PostgreSQL chart requires PV support on underlying infrastructur
+* `extraVolumeMounts`
+* `extraVolumes`
+* `livenessProbe`
+* `readinessProbe`
+
+Additionally, custom labels and annotations can be set on various resources the values of which being passed through `tpl` as well.
 
 It is important that these values be configured as strings.
-Otherwise, installation will fail. See example for Google Cloud Proxy or default affinity configuration in `values.yaml`.
+Otherwise, installation will fail.
+See example for Google Cloud Proxy or default affinity configuration in `values.yaml`.
+
+### JVM Settings
+
+Keycloak sets the following system properties by default:
+`-Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true`
+
+You can override thes by setting the `JAVA_OPTS` environment variable.
+Make sure you configure container support.
+This allows you to only configure memory using Kubernetes resources and the JVM will automatically adapt.
+
+```yaml
+extraEnv: |
+  - name: JAVA_OPTS
+    value: >-
+      -XX:+UseContainerSupport
+      -XX:MaxRAMPercentage=50.0
+      -Djava.net.preferIPv4Stack=true
+      -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS
+      -Djava.awt.headless=true
+```
 
 ### Database Setup
 
-By default, Keycloak uses an embedded H2 database.
-This is only suitable for testing purposes.
-All data is lost when Keycloak is shut down.
-Optionally, the [PostgreSQL](https://github.com/kubernetes/charts/tree/master/stable/postgresql) chart is deployed and used as database.
+By default, Bitnami's [PostgreSQL](https://github.com/bitnami/charts/tree/master/bitnami/postgresql) chart is deployed and used as database.
 Please refer to this chart for additional PostgreSQL configuration options.
 
 #### Using an External Database
 
-The Keycloak Docker image supports PostgreSQL, MySQL, MariaDB, and H2.
-The password for the database user is read from a Kubernetes secret.
-It is possible to specify an existing secret that is not managed with this chart.
-The key in the secret the password is read from may be specified as well (defaults to `password`).
+The Keycloak Docker image supports various database types.
+Configuration happens in a generic manner.
 
-```yaml
-keycloak:
-  persistence:
+##### Using a Secret Managed by the Chart
 
-    # Disable deployment of the PostgreSQL chart
-    deployPostgres: false
+The following examples uses a PostgreSQL database with a secret that is managed by the Helm chart.
 
-    # The database vendor. Can be either "postgres", "mysql", "mariadb", or "h2"
-    dbVendor: postgres
+```yaml
+postgresql:
+  # Disable PostgreSQL dependency
+  enabled: false
+
+extraEnv: |
+  - name: DB_VENDOR
+    value: postgres
+  - name: DB_ADDR
+    value: mypostgres
+  - name: DB_PORT
+    value: 5432
+  - name: DB_DATABASE
+    value: mydb
+
+extraEnvFrom: |
+  - secretRef:
+      name: '{{ include "keycloak.fullname" . }}-db'
+
+secrets:
+  - nameSuffix: db
+    stringData:
+      DB_USER: '{{ .Values.dbUser }}'
```

Comment:
Wouldn't secrets also have to be a yaml string literal (`|`) in order to pass templating like this (also example below `my-test-annotation: Test secret for {{ include "keycloak.fullname" . }}`)?

Reply:
For secrets, the individual values are passed through `tpl`.

See https://github.com/codecentric/helm-charts/pull/250/files#diff-78f4972573ad8f1af134676ca6416fb4R25

Reply:
Ah right, yep 👍  thanks